SIT.Manager.Avalonia/ViewModels/MainViewModel.cs

```diff
@@ -6,41 +6,68 @@
 using CommunityToolkit.Mvvm.Messaging;
 using FluentAvalonia.UI.Controls;
 using FluentAvalonia.UI.Media.Animation;
-using SIT.Manager.Avalonia.Interfaces;
 using SIT.Manager.Avalonia.Models;
 using SIT.Manager.Avalonia.Models.Messages;
 using System;
 using System.Collections.ObjectModel;
 using System.Diagnostics;
 using System.IO;
 using System.Threading.Tasks;
+using System.Net.Http;
+using System.Reflection;
+using Microsoft.Extensions.Logging;
 
 namespace SIT.Manager.Avalonia.ViewModels;
 
 public partial class MainViewModel : ViewModelBase, IRecipient<PageNavigationMessage>
 {
+    private const string MANAGER_VERSION_URL = @"https://raw.githubusercontent.com/stayintarkov/SIT.Manager.Avalonia/master/VERSION";
     private readonly IActionNotificationService _actionNotificationService;
     private readonly IBarNotificationService _barNotificationService;
+    private readonly ILogger<MainViewModel> _logger;
+    private readonly HttpClient _httpClient;
 
     private Frame? contentFrame;
 
     [ObservableProperty]
     private ActionNotification? _actionPanelNotification = new(string.Empty, 0, false);
 
+    [ObservableProperty]
+    private bool _updateAvailable = false;
+
     public ObservableCollection<BarNotification> BarNotifications { get; } = [];
 
     public IAsyncRelayCommand UpdateButtonCommand { get; }
 
-    public MainViewModel(IActionNotificationService actionNotificationService, IBarNotificationService barNotificationService) {
+    public MainViewModel(IActionNotificationService actionNotificationService, IBarNotificationService barNotificationService, ILogger<MainViewModel> logger, HttpClient httpClient) {
         _actionNotificationService = actionNotificationService;
         _barNotificationService = barNotificationService;
+        _logger = logger;
+        _httpClient = httpClient;
 
         _actionNotificationService.ActionNotificationReceived += ActionNotificationService_ActionNotificationReceived;
         _barNotificationService.BarNotificationReceived += BarNotificationService_BarNotificationReceived;
 
         WeakReferenceMessenger.Default.Register(this);
 
         UpdateButtonCommand = new AsyncRelayCommand(UpdateButton);
+        _ = CheckForUpdate();
```

Comment:
Possibly more of a personal thing than anything, but I'm not the biggest fan of just calling the async task in the constructor like this. 

Two ways I think I'd be happier with this is we don't run the function on creation but when the view is actually activated similar to how we do things in the ModsPageViewModel.  Or the alternative would be to use something like this `RxApp.TaskpoolScheduler.Schedule(FetchReleases);`.